clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -200,7 +200,7 @@ public synchronized boolean containsTopic(String topic) {
      * Updates the cluster metadata. If topic expiry is enabled, expiry time
      * is set for topics if required and expired topics are removed from the metadata.
      */
-    public synchronized void update(Cluster cluster, long now) {
+    public synchronized void update(Cluster cluster, boolean hasUnavailablePartitions, long now) {
```

Comment:
@hachikuji @ijuma I have updated the code to pass `unavailableTopics` in `Metadata.update` and added some more tests. The reason I didn't want to add the unavailable set to `Cluster` was because the set includes non-existent topics. Cluster felt like an odd place to store these - compared to processing them in the consumer and discarding the collection. I also thought `Cluster` was a public class, so didn't want to add something with little value to anything other than `Consumer`. I don't mind either way and am happy to change to keep it simpler.